src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl/impl/InternalEvaluatorImpl.kt

```diff
@@ -124,4 +158,73 @@ internal class InternalEvaluatorImpl(
             isExecuting = false
         }
     }
+
+    private fun updateVariablesState(cellId: Int) {
+        // remove known modifying usages in this cell
+        usageMap[cellId]?.removeIf {
+            containingMap[it] != cellId
+        }
+
+        variablesMap.forEach {
+            val state = it.value
+            val property = state.propertyKlass
+            val newValue = property.get(state.scriptInstance) ?: return@forEach
+
+            if (state.stringValue != newValue.toString()) {
+                usageMap[cellId]?.add(it.key)
+                state.stringValue = newValue.toString()
+            }
+        }
+    }
+
+    private fun getVisibleVariables(target: ResultValue, cellId: Int): Map<String, VariableState> {
+        val klass = target.scriptClass ?: return emptyMap()
+        val cellKlassInstance = target.scriptInstance!!
+
+        val fields = klass.declaredMemberProperties
+        val ans = mutableMapOf<String, VariableState>()
+        fields.forEach {
+            val casted = it as KProperty1<Any, *>
```

Comment:
There is no word `casted`. Moreover, if you simply write `it as T`, `it` will have type `T` in the following lines due to smartcasting

Reply:
But I suggest to use named lambda parameter here

Reply:
Oh, yeah, forgot about that casting context changer. And named parameter is also cool